lib/liquid/context.rb

```diff
@@ -27,7 +27,7 @@ def initialize(environments = {}, outer_scope = {}, registers = {}, rethrow_erro
       @this_stack_used = false
 
       if rethrow_errors
-        self.exception_handler = ->(e) { raise }
+        self.exception_handler = ->(e) { raise e }
```

Comment:
yeah that's what I thought too. Not sure about the semantics if it's with a lambda, but `raise e` and `raise` is usually not equivalent (one re-raises, the other raises a new exception (with a new backtrace)).